File: Level.1/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_set>

using namespace std;

string solution(vector<string> participant, vector<string> completion)
{
    unordered_multiset<string> part(participant.begin(), participant.end());

    for (const auto &e : completion)
    {
        part.erase(part.find(e));
    }

    return *part.begin();
}

File: Level.1/완주하지 못한 선수.js
function solution(participant, completion) {
    participant = participant.reduce((acc, e)=>{
        if(acc[e]){
            acc[e]++;
        } else {
            acc[e]=1;
        }
        return acc;
    }, {});
    
    completion.map(e=>{
        if(participant[e]==1){
            delete participant[e];
        } else {
            participant[e]--;
        }
    });
    
    return Object.keys(participant)[0];
}
